### ISSUE TITLE
no-unknown-animations: Add ignoreAnimations secondary option for cross-file @keyframes

### ISSUE THREAD

**Issue by shimscho** [stylelint/stylelint#9184]
### What is the problem you're trying to solve?

The `no-unknown-animations` rule only considers `@keyframes` defined within the same source file as "known". In modern CSS architectures using **design token systems**, animations are defined centrally in a shared tokens file (e.g. `tokens/animations.css`) and referenced across dozens of component files.

This makes the rule unusable without disabling it entirely — every cross-file animation reference produces a false positive.

**Example architecture:**

```
design-system/
  tokens/animations.css       ← @keyframes spin, fade-in, slide-up defined here
  primitives/
    buttons/button.css         ← animation: spin 0.6s linear infinite;
    collapse/collapse.css      ← animation: spin 0.8s linear infinite;
    file-upload/upload.css     ← animation: spin 1s linear infinite;
```

All three component files trigger `no-unknown-animations` because `spin` is not defined in the same file.

### What solution would you like to see?

A secondary option `ignoreAnimations` (similar to `ignoreAtRules` on `at-rule-no-unknown`) that accepts an array of strings or regex patterns:

```json
{
  "no-unknown-animations": [true, {
    "ignoreAnimations": ["spin", "fade-in", "/^slide-/"]
  }]
}
```

This would allow projects with centralized `@keyframes` definitions to whitelist known animation names while still catching typos in other animations.

### What alternatives have you considered?

1. **Disabling the rule globally** — loses all value
2. **Using `overrides` with file glob patterns to set the rule to `null` for specific directories** — works with the CLI, but the vscode-stylelint extension does not respect these overrides (filed separately)
3. **Duplicating `@keyframes` in every file** — defeats the purpose of centralized tokens

### Additional context

- Stylelint 17.6.0
- The rule currently accepts only a boolean — no secondary options at all
- This pattern (centralized design tokens with shared animations) is increasingly common with CSS custom property / design system architectures
- Related closed issue: #7638 (different request — about non-English property names)

**Comment by jeddy3** [comment]
Thanks for the request. Good news, there's a PR in progress for this:
- https://github.com/stylelint/stylelint/pull/9179

If that will work for your use case, I think we can close as a dupe of [#4088](https://github.com/stylelint/stylelint/issues/4088) rather add an `ignoreAnimations` secondary option?
